ReactNativeClient/lib/joplin-database.js

```diff
@@ -735,6 +735,73 @@ class JoplinDatabase extends Database {
 				);
 			}
 
+			if (targetVersion == 31) {
+				queries.push('DROP TRIGGER IF EXISTS notes_fts_before_update');
+				queries.push('DROP TRIGGER IF EXISTS notes_fts_before_delete');
+				queries.push('DROP TRIGGER IF EXISTS notes_after_update');
+				queries.push('DROP TRIGGER IF EXISTS notes_after_insert');
+
+				queries.push('DROP INDEX IF EXISTS notes_normalized_id');
+				queries.push('DROP TABLE IF EXISTS notes_normalized');
+				queries.push('DROP TABLE IF EXISTS notes_fts');
+
+
+				const notesNormalized = `
+					CREATE TABLE notes_normalized (
+						id TEXT NOT NULL,
+						title TEXT NOT NULL DEFAULT "",
+						body TEXT NOT NULL DEFAULT "",
+						user_created_time INT NOT NULL DEFAULT 0,
+						user_updated_time INT NOT NULL DEFAULT 0,
+						is_todo INT NOT NULL DEFAULT 0,
+						todo_completed INT NOT NULL DEFAULT 0,
+						parent_id TEXT NOT NULL DEFAULT "",
+						latitude NUMERIC NOT NULL DEFAULT 0,
+						longitude NUMERIC NOT NULL DEFAULT 0,
+						altitude NUMERIC NOT NULL DEFAULT 0
+					);
+				`;
+
+				queries.push(this.sqlStringToLines(notesNormalized)[0]);
+
+				queries.push('CREATE INDEX notes_normalized_id ON notes_normalized (id)');
+
+				const newVirtualTableSql = `
+					CREATE VIRTUAL TABLE notes_fts USING fts4(
+						content="notes_normalized",
+						notindexed="id",
+						notindexed="user_created_time",
+						notindexed="user_updated_time",
+						notindexed="is_todo",
+						notindexed="todo_completed",
+						notindexed="parent_id",
+						notindexed="latitude",
+						notindexed="longitude",
+						notindexed="altitude",
```

Comment:
Why are these fields not indexed? I guess we could have queries like `type:todo`, in which case an indexed `is_todo` would be useful, or do you handle this differently?

Reply:
Yes, I guess we could match "0" and "1" for `type:todo` and `type:note`, but it wouldn't work for the other fields because of '<' and '>=' comparisons.